README.md

```diff
@@ -11,6 +11,103 @@ go get github.com/moolitayer/golang/glog # Dependency
 go get github.com/moolitayer/awx-client-go/awx
 ```
 
+## Usage
+### import
+```
+import 	"github.com/moolitayer/awx-client-go/awx"
+```
+
+### Creating a connection:
+```
+# Uses the builder pattern:
+connection, err := awx.NewConnectionBuilder().
+  Url("http://awx.example.com/api").
+  Username(username).
+  Password(password).
+  Token("TOKEN").
+  Bearer("BEARER").
+  CAFile("/etc/pki/tls/cert.pem").
+  Insecure(insecure).
+  Proxy(http://myproxy.example.com).
+  Build()                                      # Create the client
+if err != nil {
+  panic(err)
+}
+defer connection.Close()                      # Don't forget to close the connection!
+```
+`Url()` points at an AWX server's root API endpoint.
+`Proxy()` specifies A proxy server to use for all outgoing connection to the AWX server.
+#### Authentication
+`Username()` and `Password()` specify Basic Auth for AWX API server.
+`Token()` uses the authtoken/ endpoint and works with AWX XXX and Ansible tower < 3.3.
+`Bearer()` uses OAuth2 and works since AWX XXX and Ansible Tower > 3.3.
+
+#### TLS
+`CAFile()` specifies a trusted PEM encoded CA certificates used to verify the AWX server.
+`CAFile()` can be used multiple times to specify a list of files.
+`Insecure(true)` can be specified to ignore TLS verification.
+
+### Supported resources
+- Projects
+- Jobs
+- Job Templates
+
+### Retrieving resources
+```
+projectsResource := connection.Projects()
+
+// Get a list of all Projects.
+getProjectsRequest := projectsResource.Get()
+getProjectsResponse, err := getProjectsRequest.Send()
```

Comment:
I don't particularly like the `.Send()` pattern, but I see it's used for `.Filter()`... How do other Go projects do this? is this a common pattern?